src/dftbp/include/error.fypp

```diff
@@ -105,5 +106,24 @@
 #:enddef PROPAGATE_ERROR
 
 
+#! Macro to fill an variable with NaN
+#:def NAN_A_VARIABLE(var)
+  block
+    use, intrinsic :: ieee_arithmetic, only : ieee_value, ieee_quiet_nan
+    ${var}$ = ieee_value(1.0_dp, ieee_quiet_nan)
+  end block
+#:enddef NAN_A_VARIABLE
+
+
+#! Macro to fill an allocatable array with IEEE NaN
+#:def NAN_AN_ALLOCATABLE(var)
+  block
+    use, intrinsic :: ieee_arithmetic, only : ieee_value, ieee_quiet_nan
+    if (allocated(${var}$)) then
+      ${var}$ = ieee_value(1.0_dp, ieee_quiet_nan)
+    end if
+  end block
+#:enddef NAN_AN_ALLOCATABLE
+
```

Comment:
As suggested above, we should probably resist the temptation to introduce macros for something which can be trivially done in Fortran.